server/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -69,11 +75,17 @@
     GetResult() {
     }
 
-    public GetResult(String index, String type, String id, long version, boolean exists, BytesReference source,
-                     Map<String, DocumentField> fields) {
+    public GetResult(String index, String type, String id, long seqNo, long primaryTerm, long version, boolean exists,
+                     BytesReference source, Map<String, DocumentField> fields) {
         this.index = index;
         this.type = type;
         this.id = id;
+        this.seqNo = seqNo;
+        this.primaryTerm = primaryTerm;
+        assert (seqNo == SequenceNumbers.UNASSIGNED_SEQ_NO && primaryTerm == 0) || (seqNo >= 0 && primaryTerm >= 1) :
```

Comment:
should we add a new UNASSIGNED_PRIMARY_TERM constant? There are too many of these magic 0 terms sprinkled all over.

Reply:
+1 here too, for a long time. I don't want to pollute this PR as it will be big and holds to more places. I can do it as a follow up.

Reply:
ok as a follow-up